### DISCUSSION TITLE
Standups - Sprint 3

### DISCUSSION THREAD

**Discussion by JeanPierreNell** [tazama-lf/General-Issues#112] [General]
## Attendees
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@kawaki-san - Rodney Kanjala

## Apologies
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@vorsterk - Kyle Vorster

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Worked on Rule 18 , 21 and 1. Merge 18 into Main Branch.  Rule 007 moved to QA for Justus to Test. Investigate issue with the levenshtein distance library. Rule 21 pending JP

Today: Continue Rule 007 and Rule 001. 

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: Optimised Rule 054 for processing transactions on Sandbox 2. Played around with time delay. Continue with sending relevant data using the data Cache property. (issue #76)

Today: Testing of Issue #76 with use cases. Assist with issues from team

**@kawaki-san - Rodney Kanjala**

Yesterday: Continue on Rule 34, 35 , 36. Waiting on Len for updates. Making branch in the library for new Arango Calls.

Today: Finish up 34, 35 and 36. Using Generic functions. 

## Parking lot

Follow up on demo videos: No Demo videos required today.

**Comment by JeanPierreNell** [comment]
## 2023/06/06

## Attendees
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@kawaki-san - Rodney Kanjala

## Apologies
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@vorsterk - Kyle Vorster

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday : Worked on Rule-007 started afresh to integrate the library and stared re-factorizing rule-001 and made the demo video

Today : Continue with rule-001 and testing rule-007 to make sure that rule-007 works with even the transaction provided by Justus

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: Worked on Testing for Data Preperation. Optimisation of Rules being executed.

Today: Finishing up Data Prep Changes with tests.

**@kawaki-san - Rodney Kanjala**

Yesterday: Continued Refactor on Rule 25 with testing , 17 and 11. End to end testing is up to shape. Continued with Rule 34, using the generic Query. Data between the Generic and base queries are different. 

Today: Investigate data difference between Queries. Continue with Rule 34 , 35 and 36. 

## Parking lot

**Comment by JeanPierreNell** [comment]
## 2023/06/07

## Attendees
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@kawaki-san - Rodney Kanjala

## Apologies
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@vorsterk - Kyle Vorster

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Worked on Rule 1. Almost done, on clean-up step. Attended to issues JP Pointed out for Rule 021. Forbidden Errors on Jenkins, will reach out to JP if stuck.
Today: Create PR Today + Request Review + Set-Up Jenkins Job. Worked on Rule 27.

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: Finished working on the data prep service, adding caching to requests. Tests done. 
Today: Ask Foley for priority of next work, continue with sending batches to Sandbox2 

**@kawaki-san - Rodney Kanjala**

Yesterday: Worked on Rule 25. Continued on Rule 34, one scenario on postman is failing. 3/4 Tests is passing. 
Today: Work on Rule 35 and 36.

## Parking lot

Rodney: Postman Beta Access Requested. It has performance benchmarking included - compared Rust performance with Node.js.

**Comment by JeanPierreNell** [comment]
## 2023/06/08

## Attendees
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@kawaki-san - Rodney Kanjala

## Apologies
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@vorsterk - Kyle Vorster

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Merged Rule 1. Rodney helped with issues on jenkins - token expired but worked on locally. Worked on Rule 27. Problem with the methods. Len's refactor removed parameters. Rule 27 is done with husky, lint and prettier. 

Today: Continue with Rule 027, if done will continue on. 

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: Worked on Issue with Typology Processor using the wrong configuration value. Continued with sending through batches. Received the NATS library from Foley. 

Today: Work on understanding and implementing the NATS startup lib

**@kawaki-san - Rodney Kanjala**

Yesterday: Sorted out Jenkin issue with Shezi and investigate functionality of the library. Worked on Rule 35. Proceed with Rule 044. Finished, opening PR Step. Resolved Merge on rule 025 from Jp. 

Today: Will continue with Rule 044. Will delegate with Shezi.

## Parking lot

**Comment by JeanPierreNell** [comment]
## 2023/06/09

## Attendees
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@kawaki-san - Rodney Kanjala

## Apologies
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@vorsterk - Kyle Vorster

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Discovered Function on Rule 27 is miss-labeled. Tried to adjust the AQL and run directly on Arango - AQL is wrong. Checked the Library for other compatible Queries. The previous Query was Filtering wrong. Refactored Postman Test for sanity. PR was send to Rodney. Rule 27 is Merged. Moved on to Rule 10. Checking Husky.

Today: Continue with Rule 010. Will ask for PR review.

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: Worked on Implementing NATS + Sending Batches. Will require 2 processors to effectively test.

Today: Continue on Implementing NATS + Sending Batches.

**@kawaki-san - Rodney Kanjala**

Yesterday: Worked on Rule 044. Reviewed Shezi's PR on Rule 027. PR was opened for Rule 044. Changes was mentioned by Shezi. Removed Arango dependancies. Worked on Rule 016, will require PR review. Will delegate Rule 40 onwards with Shezi.

Today: Continue working on Rule 016, then continue with Rule 045. 

## Parking lot

**Comment by Justus-at-Tazama** [comment]
## 2023/06/12

## Attendees
Jason Darmanovich
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@kawaki-san - Rodney Kanjala

## Apologies
@johanfol - Johan Foley

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Completed rule 010 and rule 045. PR's to go up for review.

Today: Continue with rule 048.

**@vorsterk - Kyle Vorster**

Yesterday: Code review

Today: Try to figure out why we are not reaching our TPS.

**@Lenbkr - Len Bekker**

Yesterday: Code review

Today: Uncertain. Performance assessment on Arango.

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: Data preparation service is completed with the NATS implementation. The second service is required to connect two processors. The TMS will ingest the http message and then pass it to the data preparation service. A specific user story is required to guide the work. Pause work on the beta testing.

Today: Scope and define the work for the NATS implementation. Carrying on with the NATS implementation.

**@kawaki-san - Rodney Kanjala**

Yesterday: Completed rules 044 and rule 016 - sent up the PRs. 

Today: Rule 044 and rule 016 PR review feedback. Checking if the Jenkins build is passing.

**@johanfol - Johan Foley**

Yesterday: Flying

Today: Landing

## Parking lot

Brainstorming today.

**Comment by Justus-at-Tazama** [comment]
## 2023/06/05

## Attendees
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@kawaki-san - Rodney Kanjala
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@vorsterk - Kyle Vorster

## Apologies

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Completed the last two rules. Also refactored rule 084 simil.ar to rule 083. PR's created this morning and reviewed by Rodney. Walked through rule 011.

Today: Review rule 011.

**@Lenbkr - Len Bekker**

Yesterday: Investigating connection pooling and database optimisation. Tweaking Arango driver settings and exploring various options. 30 to 40% faster on http calls. Lots of weirdness.

Today: Continue optimisation.

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: Reviewed the Redis interaction. You can use one of two methods. The functions were incorrectly named. Functions renamed and new functions added. Testing on data preparation stuff - everything is working, even an end-to-end test. Ran into an issue with the build and deployment that Kyle is supporting. Mopped up batch PPA merge to populate main branch. PoC state only.

Today: Continue with NATS. Trying to communicate between TMS and DPS.

**@kawaki-san - Rodney Kanjala**

Yesterday: Took on rule 084, made progress, got stuck, gave it back. Reviewed rule 011.

Today: Review rule 011

**@vorsterk - Kyle Vorster**

Yesterday: working on the jobs for Jenkins. Added all the workflows for the release yaml files. Created a Jenkins job to run it. Getting authentication issues. Needs to set up a github app for Jenkins, but this is already set up. This is a delightful puzzle. It's failing github publish checks.

Today: Carrying on with the Jenkins build - should have it up today. Then carrying on to the rest of the deployment.
The data preparation testing is not using the version of the library that we are using. Kyle supporting JP to resolve this issue to help him check in his code. The code needs to "package" to be deployed. Foley suggests to check in the GitHub Actions stage to see if the account can be impersonated.

**@johanfol - Johan Foley**

Yesterday: Got the library up and running for the scaffolding.

Today: Carrying on with this work. Hope to have the template up and running today.

## Parking lot

Buggy rules: Log a bug - please also check APM logging on SB4, as well as deployment of the most recent code to SB4.

Story 111: Assign to Foley and link to template work.

Rule priorities: Bugged rules first.

Redis vs tests: Testing for common centralised functions such as DetermineOutcome should be handled in their new origin.
